pallets/auctions/src/lib.rs

```diff
@@ -0,0 +1,726 @@
+                    auction.start_block < current_block,
+                    Error::<T>::AuctionNotStarted
+                );
+
+                // ensure the auction period has not ended
+                ensure!(auction.end_block > current_block, Error::<T>::AuctionEnded);
+                auction.state = AuctionState::Completed;
+
+                // transfer NFT to caller
+                T::NFTHandler::set_owner(nft_id, &who)?;
+                // Mark NFT as not listed for sale
+                T::NFTHandler::set_listed_for_sale(nft_id, false)?;
+
+                // get marketplace fee for transaction
+                let marketplace_info =
+                    T::MarketplaceHandler::get_marketplace_info(auction.marketplace_id)
+                        .ok_or(Error::<T>::UnexpectedError)?;
+                let marketplace_commission_amount =
+                    amount.saturating_mul(marketplace_info.1.into()) / 100u32.into();
+                let amount_to_auction_creator =
+                    amount.saturating_sub(marketplace_commission_amount);
+
+                // Transfer fee to marketplace
+                T::Currency::transfer(
+                    &Self::account_id(),
+                    &marketplace_info.0,
+                    marketplace_commission_amount,
+                    KeepAlive,
+                )?;
+
+                // Transfer remaining to auction creator
+                T::Currency::transfer(
+                    &Self::account_id(),
+                    &auction.creator,
+                    amount_to_auction_creator,
+                    KeepAlive,
+                )?;
+
+                // emit bid created event
+                Self::deposit_event(Event::AuctionBuyItNow {
+                    nft_id,
+                    marketplace_id: auction.marketplace_id,
+                    creator: who,
+                });
+
+                Ok(())
+            })?;
+
+            Ok(().into())
+        }
+
+        #[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
+        pub fn complete_auction(origin: OriginFor<T>, nft_id: NFTId) -> DispatchResultWithPostInfo {
+            let _who = ensure_root(origin)?;
+            let current_block = frame_system::Pallet::<T>::block_number();
+
+            // Mark the auction as completed so other bid users can claim amount
+            Auctions::<T>::try_mutate(nft_id, |maybe_auction| -> DispatchResult {
+                // should not panic when unwrap since already checked above
+                let mut auction = maybe_auction
+                    .as_mut()
+                    .ok_or(Error::<T>::AuctionDoesNotExist)?;
+
+                // ensure the auction period has ended
+                ensure!(
+                    auction.end_block < current_block,
+                    Error::<T>::AuctionNotCompleted
+                );
+
+                // get the highest bidder
+                let highest_bidder = auction.bidders.get_highest_bid().cloned();
```

Comment:
Instead of getting the same bidder twice, you can replace this call with auction.bidders.remove_bid(0).